src/components/Header/index.jsx

```diff
@@ -1,102 +1,120 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
-import { Button, Icon, Dropdown } from '@edx/paragon';
+import {
+  Dropdown, Navbar, AvatarButton, Nav,
+} from '@edx/paragon';
 
+import {
+  getAuthenticatedUser, hydrateAuthenticatedUser, getLogoutRedirectUrl,
+} from '@edx/frontend-platform/auth';
 import SidebarToggle from '../../containers/SidebarToggle';
 import Img from '../Img';
 
-import apiClient from '../../data/apiClient';
 import { configuration } from '../../config';
 
 import './Header.scss';
 
-class Header extends React.Component {
-  componentDidMount() {
-    if (this.props.username) {
-      this.props.fetchUserAccount(this.props.username);
-    }
-  }
+export const Logo = ({ enterpriseLogo, enterpriseName }) => {
+  const logo = configuration.LOGO_URL;
+
+  return (
+    <Img
+      src={enterpriseLogo || logo}
+      alt={`${enterpriseName || 'edX'} logo`}
+      onError={(e) => { e.target.src = logo; }}
+    />
+  );
+};
+
+Logo.defaultProps = {
+  enterpriseLogo: configuration.LOGO_URL,
+};
+
+Logo.propTypes = {
+  enterpriseLogo: PropTypes.string,
+  enterpriseName: PropTypes.string.isRequired,
+};
 
-  getProfileIconElement() {
-    const { email, userProfileImageUrl } = this.props;
-    const screenReaderText = `Profile image for ${email}`;
+export const HeaderDropdown = ({ user }) => {
+  const { profileImage, username } = user;
+  const avatarImage = profileImage?.hasImage ? profileImage.imageUrlMedium : null;
+  const avatarScreenReaderText = `Profile image for ${username}`;
 
-    if (userProfileImageUrl) {
-      return <Img className="user-profile-img mr-2" src={userProfileImageUrl} alt={screenReaderText} />;
+  return (
+    <Dropdown>
+      <Dropdown.Toggle
+        as={AvatarButton}
+        src={avatarImage}
+        id="avatar-dropdown"
+        size="md"
+        alt={avatarScreenReaderText}
+      >
+        {username}
+      </Dropdown.Toggle>
+      <Dropdown.Menu>
+        <Dropdown.Item
+          href={getLogoutRedirectUrl()}
```

Comment:
We should pass in the url to the branded logistration page as an argument to `getLogoutRedirectUrl` here so that the user is redirected to the branded logistration page after logout.

Ssee `getProxyLoginUrl` function.

Reply:
Ignore this. This will be tackled via follow-up work ENT-4040!